.scripts/nightly.yml

```diff
@@ -0,0 +1,29 @@
+schedules:
+  - cron: "0 0 * * *"
+    displayName: Daily midnight build
+    branches:
+      include:
+        - main
+
+steps:
+  - script: npm install -g @typespec/compiler@next
+    displayName: Install @typespec/compiler@next
+  - template: build.yml
+  - script: npm run override-dev-version
+    workingDirectory: $(Build.SourcesDirectory)/packages/typespec-ts
+    displayName: "Override pnpm-config version to latest dev"
+  - script: npx @azure-tools/typespec-bump-deps package.json
+    workingDirectory: $(Build.SourcesDirectory)/packages/typespec-ts
+    displayName: "Override package.json version to latest dev"
+  - script: rush update --full && rush build
+    workingDirectory: $(Build.SourcesDirectory)/packages/typespec-ts
+    displayName: "Re-build after update the config"
+  - script: npm list
+    workingDirectory: $(Build.SourcesDirectory)/packages/typespec-ts
+    displayName: "List the current dependency"
+  - script: npm run unit-test
+    workingDirectory: $(Build.SourcesDirectory)/packages/typespec-ts
+    displayName: "Run all unit test cases"
+  - script: npm run integration-test-ci
+    workingDirectory: $(Build.SourcesDirectory)/packages/typespec-ts
+    displayName: "Run all Cadl-ranch test cases"
```

Comment:
why not run smoke-test in typespec-test folder

Reply:
let me add that